pipenv/utils/indexes.py

```diff
@@ -82,10 +82,6 @@ def get_source_list(
             extra_src = get_project_index(project, source)
             if not sources or extra_src["url"] != sources[0]["url"]:
                 sources.append(extra_src)
-        else:
```

Comment:
How are you handling the supposed expected behavior?
Should it not be addressed?

Reply:
Yeah, I don't think this part should be removed because I think it might be what adds a new index to the `Pipfile` when specifying one that does not exist via the CLI.

Reply:
I updated the pull to simply move the code outside of the `for`/`else`.

Reply:
The change is still problematic, witthout the else you could be adding the sources twice -- I am not sure what the intent here is moving this outside the else.  There are two failing tests though:
> FAILED tests/integration/test_lock.py::test_private_index_skip_lock - assert ...
FAILED tests/integration/test_install_uri.py::test_install_specifying_index_url

Reply:
Unless I'm missing something, which I definitely could be, the `else` was always executed because the `for source in extra_indexes` loop was never terminated by a `break` statement. As a result I simply moved the second `for` loop outside the `else` statement to make it obvious that it was always executed.

https://docs.python.org/3/tutorial/controlflow.html#break-and-continue-statements-and-else-clauses-on-loops

I think the test failures are caused by a different commit on this pull.

Reply:
I see, this change makes sense then.